### PR TITLE
Add event retry logic

### PR DESCRIPTION
**Pull request by aliokn**
Added event retry logic to producer side and also created a dead letter topic for failed consumer events

### CODE REVIEW

File: src/EventBus.Kafka/KafkaEventBus.cs [hunk 1]
```diff
@@ -152,18 +172,93 @@ public async Task CreateTopicAsync(string eventName)
...
+            throw;
+        }
+    }
+
+    private async Task CreateDeadLetterTopic()
+    {
+        try
+        {
+            IAdminClient adminClient = new AdminClientBuilder(this._producerConfig).Build();
+            List<TopicMetadata>? topics = adminClient.GetMetadata(TimeSpan.FromSeconds(10)).Topics;
```
Comment (SalihKARAHAN) [inline src/EventBus.Kafka/KafkaEventBus.cs:225]: We should define the value 10 as const with a meaningful name

File: src/EventBus.Kafka/KafkaEventBus.cs [hunk 2]
```diff
@@ -12,24 +14,31 @@ namespace FonRadar.Base.EventBus.Kafka;
 
 public class KafkaEventBus : IKafkaEventBus
 {
+    private const string DEAD_LETTER = "DeadLetter";
```
Comment (SalihKARAHAN) [inline src/EventBus.Kafka/KafkaEventBus.cs:17]: You should change the name DEAD_LETTER_TOPIC_NAME

**Review by SalihKARAHAN — CHANGES_REQUESTED** [review]

File: src/EventBus.Kafka/ServiceRegistrationExtension.cs [hunk 3]
```diff
@@ -55,7 +55,7 @@ public static class ServiceRegistrationExtension
         serviceCollection.AddTransient<KafkaServiceConfigurationValidator>();
         serviceCollection.AddSingleton<TEventBusType, TEventBusImplementationType>(provider =>
         {
-            ILogger<IEventBus> logger = provider.GetRequiredService<ILogger<IEventBus>>();
+            ILogger<IEventBus> logger = provider.GetRequiredService<ILogger<KafkaEventBus>>();
```
[the remark this answers is not in the document]
Comment (aliokn) [reply src/EventBus.Kafka/ServiceRegistrationExtension.cs:58]: Yeah you are right, haven't pushed the latest one yet

File: src/EventBus.Kafka/KafkaEventBus.cs (hunk 1 above)
Comment (aliokn) [reply src/EventBus.Kafka/KafkaEventBus.cs:225]: Would a constant be OK ?

Comment (SalihKARAHAN) [reply src/EventBus.Kafka/KafkaEventBus.cs:225]: Already it is a const value, we should move it to a friendly name to reach the clean code.

Comment (aliokn) [reply src/EventBus.Kafka/KafkaEventBus.cs:225]: Done

**Review by SalihKARAHAN — APPROVED** [review]
Excellent!

**Pull request merged**